### PR TITLE
Improve Contacts Manager's related documentation

### LINKED ISSUE #55: Update Contacts Manager v2.0 features on UG

### PR DESCRIPTION
**Pull request by wli-linda**
Resolves #55 and #79

### CODE REVIEW

File: docs/diagrams/ContactSetFieldsSequence.puml [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-participant ":Field"
+box LOGIC_COLOR_T1
+participant "contactsManager:ContactsManager" LOGIC_COLOR
+participant ":Field" LOGIC_COLOR
+participant "contact:Contact" LOGIC_COLOR
+end box
+
+
+box  MODEL_COLOR_T1
+participant  "ContactParser" <<class>> MODEL_COLOR
```
Comment (OzairHasan) [inline docs/diagrams/ContactSetFieldsSequence.puml:12]: Good updates to the existing sequence diagrams.

File: src/main/java/seedu/allonus/contacts/ContactParser.java [hunk 2]
```diff
@@ -51,11 +62,18 @@ static int parseNum(String userInput) {
...
+            logger.log(Level.FINE, String.format(CONTACTS_PARSER_INVALID_FIELD_LOG_MESSAGE, fieldString));
             throw new InvalidContactField(CONTACTS_PARSER_INVALID_FIELD_MESSAGE);
         }
     }
 
+    /**
+     * Updates the fields of a given Contact object.
+     *
+     * @param contact The Contact object to be updated.
+     * @param fieldStrings An ArrayList of strings of fields to be updated.
```
Comment (OzairHasan) [inline src/main/java/seedu/allonus/contacts/ContactParser.java:74]: Excellent use of javadoc headers

File: src/main/java/seedu/allonus/contacts/ContactsManager.java [hunk 3]
```diff
@@ -178,7 +209,7 @@ public static void contactsRunner(TextUi ui) {
                 editContact(userInput);
             } else {
                 printFormat(CONTACTS_INVALID_COMMAND_MESSAGE);
-                logger.log(Level.FINER, String.format("Invalid command to Contacts Manager: %s", userInput));
+                logger.log(Level.FINER, String.format(CONTACTS_INVALID_COMMAND_LOG_MESSAGE, userInput));
```
Comment (OzairHasan) [inline src/main/java/seedu/allonus/contacts/ContactsManager.java:212]: I think for the final iteration, we should standardize how we log our messages.

**Review by OzairHasan — COMMENTED** [review]
LGTM

File: src/main/java/seedu/allonus/contacts/ContactsManager.java [hunk 4]
```diff
@@ -31,9 +40,19 @@
...
+            "\nNow you have %d contacts in the list.";
+
+    public static final String CONTACTS_FIND_EMPTY_KEYWORD_MESSAGE =
+            "You need to specify the keyword you want to find!";
+    public static final String CONTACTS_FIND_MULTIPLE_KEYWORDS_MESSAGE =
+            "Please only enter one keyword!";
+    public static final String CONTACTS_FIND_NO_MATCHES_MESSAGE =
+            "There are no contacts matching this keyword!";
+    public static final String CONTACTS_FIND_SUCCESS_MESSAGE =
+            "Here are the matching contacts in your list:\n";
```
Comment (YitHien) [inline src/main/java/seedu/allonus/contacts/ContactsManager.java:55]: Nice refactoring of string constants!

**Review by YitHien — COMMENTED** [review]
Great code overall! LGTM

File: text-ui-test/input.txt [hunk 5]
```diff
@@ -5,6 +5,14 @@ add n/Daniela Koh d/met at SoC canteen e/other_email@u.nus.edu f/FoS
 list
 find Daniela
 find Koh
+find
```
Comment (mathanmahe) [inline text-ui-test/input.txt:8]: Good update of your test cases at every feature to better deal with regression testing.

**Review by mathanmahe — APPROVED** [review]
Good  improvements on the documentation. LGTM!

File: src/main/java/seedu/allonus/contacts/ContactsManager.java (hunk 3 above)
Comment (wli-linda) [reply src/main/java/seedu/allonus/contacts/ContactsManager.java:212]: Yes agreed, thanks for pointing this out

**Pull request merged**